### PR TITLE
docs: add typography section to heading

### PR DESCRIPTION
**Pull request by richbachman**
Proposed solution for not having a `Typography` component page on the docs site. Page is unnecessary at this point since it mainly just links to the `Heading` and `Paragraph` pages.

Similar text will need to be added to `Paragraph` and `List` pages when ready.

- [x] Added Typography component paragraph to `Heading`
- [x] Removed `Typography` component from left nav and component list

### CODE REVIEW

**Comment by richbachman** [comment]
@aayushpi @serifluous @two24studios will you all take a look at this copy I added to the `Heading` page referencing the `Typography` component, and let me know if that works? Thanks!

**Comment by serifluous** [comment]
+1 to @two24studios's edit.

This section could make more sense placed between the "Installation" and "Usage" sections.

**Comment by richbachman** [comment]
@serifluous What about as part of Composition notes? I'm mentioning Typography component install instructions in the install and usage sections.

**Comment by serifluous** [comment]
It's more that the API section could use more explanation about what the Typography component is, because right now it just shows how to install and doesn't explain what's included.

**Comment by richbachman** [comment]
@serifluous I added it after the install section of API.

File: packages/paste-website/src/components/component-overview-table/index.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const ComponentOverviewTable: React.FC<ComponentOverviewTableProps> = ({category
     .filter(({node}) => node.status === PackageStatus.BACKLOG)
     .sort(sortNodeByName);
   const sortedComponentsList = componentsList
-    .filter(({node}) => node.status !== PackageStatus.BACKLOG)
+    .filter(({node}) => node.status !== PackageStatus.BACKLOG && node.name !== '@twilio-paste/typography')
```
Comment (TheSisb) [inline packages/paste-website/src/components/component-overview-table/index.tsx]: This line made me notice that the typography meta-package is in the components folder.  It should probably be a sibling to it, like the core package.  No? @SiTaggart thoughts appreciated here too

Comment (SiTaggart) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: Yeah, potentially actually.

Would core still include typography, heading, paragraph and list? Or (my thinking) should typography just be it's own package outside of core, and core is just a bundle of the individual components?

Comment (SiTaggart) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: Further reasoning is I noticed there's a bit of an exported name clash when you build and deploy core. 

```
core
  exports typography
    exports heading
    exports paragraph
  exports heading
  exports paragraph
```
so heading and paragraph clash (or something, forget the exact warning)

Comment (TheSisb) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: I'm starting to feel like the typography package is unnecessary complexity.  `core` and individual packages should be sufficient.

Comment (SiTaggart) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: > I'm starting to feel like the typography package is unnecessary complexity.

Now we're talking, simplify. I lean towards removing too

Comment (richbachman) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: I agree on this one. May have one been a good idea, but unnecessary now with `core`. So we need a ticket to deprecate the `typography` package? Any others?

Comment (TheSisb) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: I think just Typography.  We need to remove any references to it from other packages/docs too

Comment (richbachman) [reply packages/paste-website/src/components/component-overview-table/index.tsx]: Jira ticket created. I'm going to close this PR and open another one to remove references from other packages/docs.

**Pull request closed without merging**